plugins/content/responsiveimages/responsiveimages.php

```diff
@@ -0,0 +1,293 @@
+		if ($unusedImages = $session->get('responsiveimages.unused'))
+		{
+			$this->_displayModal($unusedImages);
+
+			// Clear session
+			$session->clear('responsiveimages.unused');
+		}
+
+		return true;
+	}
+
+	/**
+	 * Event that stores initial versions of images and inserts srcset and sizes
+	 * attributes into content img tags
+	 *
+	 * @param   string   $context  The context
+	 * @param   object   $table    The item
+	 * @param   boolean  $isNew    Is new item
+	 * @param   array    $data     The validated data
+	 *
+	 * @return  boolean
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onContentBeforeSave($context, $table, $isNew, $data)
+	{
+		// Check type of table object
+		if ($table instanceof Table)
+		{
+			$contentKey = $this->_getContentKey($context);
+
+			// Add srcset attribute to content images
+			$table->{$contentKey} = ResponsiveImagesHelper::addContentSrcsetAndSizes($table->{$contentKey});
+
+			// Get initial versions of content and form images
+			$item = clone $table;
+			$item->load($table->id);
+
+			if ($formImages = $this->_getFormImages($context, (array) $item))
+			{
+				$this->initFormImages = $formImages;
+			}
+
+			if ($content = $item->{$contentKey})
+			{
+				$this->initContent = $content;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Event that generates different sized versions of content and form images
+	 *
+	 * @param   string   $context  The context of the content passed to the plugin (added in 1.6)
+	 * @param   object   $article  A JTableContent object
+	 * @param   boolean  $isNew    If the content is just about to be created
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onContentAfterSave($context, $article, $isNew): void
+	{
+		// Check type of article object
+		if ($article instanceof Table)
+		{
+			// Generate responsive images for form and content
+			$formImages = $this->_getFormImages($context, (array) $article);
+
+			if (!is_null($formImages))
+			{
+				$unusedFormImages = ResponsiveImagesHelper::getUnusedFormImages($this->initFormImages, $formImages);
+
+				ResponsiveImagesHelper::generateFormImages($formImages);
+			}
+
+			$content = $article->{$this->_getContentKey($context)};
+
+			if (!is_null($content))
+			{
+				$unusedContentImages = ResponsiveImagesHelper::getUnusedContentImages($this->initContent, $content);
+
+				ResponsiveImagesHelper::generateContentImages($content);
+			}
+
+			$unusedImages = array_merge($unusedFormImages ?? [], $unusedContentImages ?? []);
+
+			// Set unused images in session if not empty
+			if (!empty($unusedImages))
+			{
+				Factory::getApplication()->getSession()->set('responsiveimages.unused', json_encode($unusedImages));
+			}
+		}
+	}
+
+	/**
+	 * Returns form images from data with specific context
+	 *
+	 * @param   string  $context  The context for the data
+	 * @param   array   $data     The validated data
+	 *
+	 * @return  mixed   Array of form images or null if they don't exist
```

Comment:
`mixed` and `object` are just lazy ways to document :) always try to be more specific. 

```suggestion
	 * @return  array|null   Array of form images or null if they don't exist
```